src/uint/extra_trait_impls/concat.rs

```diff
@@ -0,0 +1,449 @@
+use crate::{
+    Uint, U1024, U128, U1280, U1536, U1792, U192, U2048, U256, U3072, U320, U3584, U384, U4096,
+    U448, U512, U576, U6144, U64, U640, U768, U8192, U896,
+};
+
+use crate::{Concat, Limb};
+
+macro_rules! impl_concat_cross_sizes {
```

Comment:
Because `impl_concat` defines the following method for the type `pub const fn concat(&self, rhs: &Self) -> Uint<{nlimbs!($bits) * 2}>` which is specific for concatenating `Self` with `Self`. 

What I would have preferred was changing this signature in a similar fashion to what I did with `mul_wide`, i.e. to `pub const fn concat<const RHS_LIMBS: usize>(&self,rhs: &Uint<RHS_LIMBS>,) -> Uint<{LIMBS + RHS_LIMBS}>`.

However, seems we can't do that until `generic_const_exprs` becomes stable:
```
const parameters may only be used as standalone arguments, i.e. `RHS_LIMBS` 
use `#![feature(generic_const_exprs)]` to allow generic const expressions
```

Reply:
`impl_concat` is internal but it defines `concat` for `Uint` which is an exported method.

Reply:
Yes, so long as you don't make breaking changes to the emitted code there shouldn't be a problem